intel/intel-extension-for-transformers
# Enable "neuralchat predict" for Intel GPU

## Type of Change

Add xpu support for neuralchat predict

## Description

Enable "neuralchat predict" for Intel GPU

## Expected Behavior & Potential Risk

Expected Behavior: neuralchat predict --query "Tell me about Intel Xeon Scalable Processors." can be run on Intel PVC and Arc GPU.

Potential Risk: Only "neuralchat predict" command is tested.

## How has this PR been tested?

1. build itrex from source code

2. install packages
    cd intel_extension_for_transformers/neural_chat
    pip install -r requirements_xpu.txt
    pip install --no-deps -r requirements_xpu_no_deps.txt

3. run command
    neuralchat predict --query "Tell me about Intel Xeon Scalable Processors."
You can use xpu manager to observe status of Intel GPU during running command.

## Dependency Change?

1. In requirements_xpu.txt, only packages needed for "neuralchat predict" command are added and removed others.

2. peft is imported when neuralchat is launched, while it has compatibility issue with torch for xpu, so put it in requirements_xpu_no_deps.txt and need to install with --no-deps at the time being.
